.github/workflows/superset.yml

```diff
@@ -0,0 +1,131 @@
+name: Build Superset
+
+on:
+  # only build on `master` pushes
+  push:
+    branches: [ master ]
+  # but build on pull requests to any branch
+  pull_request:
+
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6]
+    env:
+      PYTHON_LINT_TARGET: setup.py superset tests
+      SUPERSET_CONFIG: tests.superset_test_config
+      REDIS_HOST: redis
+
+    services:
+      postgres:
+        image: postgres:10.1-alpine
+        env:
+          POSTGRES_USER: superset
+          POSTGRES_DB: superset
+          POSTGRES_PASSWORD: superset
+        ports:
+        - 5432:5432
+        # Add a health check
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+      redis:
+        image: redis:5-alpine
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: true
+          MYSQL_ROOT_PASSWORD: ''
+          MYSQL_USER: superset
+          MYSQL_DATABASE: superset
+          MYSQL_PASSWORD: superset
+    steps:
```

Comment:
It seems this will execute all tasks sequentially.

Reply:
Yes, that's one of the limitations. GitHub Actions only supports workflow level parallelization, which means each workflow would have to run the same build process again and have some duplicate code. Good news is there is cache at the workflow level, so the only thing we need to look at is how to make the build code reusable.